src/sdk/namespace_blob.js

```diff
@@ -377,35 +377,67 @@ class NamespaceBlob {
         // generating block ids in the form: '95342c3f-000005'
         // this is needed mostly since azure requires all the block_ids to have the same fixed length
         const block_id = this.blob.getBlockId(params.obj_id, params.num);
+        let res;
         dbg.log0('NamespaceBlob.upload_multipart:',
             this.container,
             inspect(_.omit(params, 'source_stream')),
             'block_id', block_id
         );
         if (params.copy_source) {
-            throw new Error('NamespaceBlob.upload_multipart: copy part not yet supported');
-        }
 
-        let count = 1;
-        const count_stream = stream_utils.get_tap_stream(data => {
-            stats_collector.instance(this.rpc_client).update_namespace_write_stats({
-                namespace_resource_id: this.namespace_resource_id,
-                size: data.length,
-                count
-            });
-            // clear count for next updates
-            count = 0;
-        });
-        const res = await P.fromCallback(callback =>
-            this.blob.createBlockFromStream(
-                block_id,
-                this.container,
-                params.key,
-                params.source_stream.pipe(count_stream),
-                params.size, // streamLength really required ???
-                callback)
-        );
+            const start = params.copy_source.ranges ? params.copy_source.ranges[0].start : 0;
+            const end = params.copy_source.ranges ? params.copy_source.ranges[0].end : params.size;
+
+            const maxCopySource = 100 * 1024 * 1024;
+            if (end - start > maxCopySource) {
+                throw new Error('The specified copy source is larger than the maximum allowable size for a copy source (100mb).');
+            }
+
+            let expiryDate = new Date();
```

Comment:
Maybe we can move this logic to cloud_utils, it seems we already used same logic in  _get_shared_access_signature in block_store_azure

Reply:
Also maybe take some time before current time like in that function - to be on the safe side.